ScraperFC/SofaScore.py

```diff
@@ -1,144 +1,454 @@
+            data = response.json()[team]['players']
+
+            for item in data:
+                player_data = item['player']
+                player_ids[player_data['name']] = player_data['id']
+
+        return player_ids
+    
+    def get_match_data(self, match_url):
+        """Get match general data
+
+        Args:
+            match_url (string): Full link to a SofaScore match
+
+        Returns:
+            dict/json: Generic data about a match in particular
+        """
+
+        match_id = self.get_match_id(match_url)
+        headers = self.get_requests_headers()
+
+        response = requests.get(f'https://api.sofascore.com/api/v1/event/{match_id}', headers=headers)
+        data = response.json()['event']
+        return data
+    
+    def get_team_names(self, match_url):
+        """Get the team names for the home and away teams
+
+        Args:
+            match_url (string): Full link to a SofaScore match
+
+        Returns:
+            strings: Name of home and away team.
+        """
+
+        data = self.get_match_data(match_url)
+
+        home_team = data['homeTeam']['name']
+        away_team = data['awayTeam']['name']
+
+        return home_team, away_team
+
+    def match_momentum_plot(self, match_momentum_df, match_id, colors = ['red', 'green']):
+        """Plot Match Momentum replicating the one on SofaScore UI
+
+        Args:
+            match_momentum_df (DataFrame): DataFrame generated in match_momentum functions. Contains two columns: Minute and value (if > 0, momentum was with home side and viceversa)
+            match_id (string): Match Id for a SofaScore match
+            colors (list, optional): Colors of the bars in the graph. Defaults to ['red', 'green'].
+
+        Returns:
+            fig, ax: A png and the fig and axes for further customization
+        """
+
+        plot_colors = [f'{colors[0]}' if value < 0 else f'{colors[1]}' for value in match_momentum_df.value]
+
+        fig,ax = plt.subplots(figsize=(16,9))
+        fig.set_facecolor('white')
+
+        ax.bar(match_momentum_df.minute, match_momentum_df.value, color=plot_colors)
+        ax.axvline(45.5, ls=':')
+        ax.set_xlabel('Minutes')
+        ax.set_xticks(range(0,91,10))
+        ax.set_xlim(0,91)
+
+        plt.gca()
+        ax.spines[['top', 'right', 'left']].set_visible(False)
+        ax.set_yticks([])
+        plt.savefig(f'{match_id}_match_momentum.png', bbox_inches='tight')
+        
+        return fig, ax
+    
+    def get_match_id(self, match_url):
+        """Get match id for any match
+
+        Args:
+            match_url (string): Full link to a SofaScore match
+
+        Returns:
+            string: Match id for a SofaScore match. Used in Urls
+        """
+        match_id = match_url.split('#')[-1]
+        return match_id
+
+    def scrape_league_stats(self, tournament, season, accumulation='total', selected_positions=['Goalkeepers', 'Defenders', 'Midfielders', 'Forwards']):
```

Comment:
Not trying to be an *ss, but can `tournament` and `season` be changed to `year` and `league` so that this fits with the rest of ScraperFC? Just want this all to feel like 1 cohesive package.